share/spack/gitlab/cloud_pipelines/configs/ci.yaml

```diff
@@ -9,11 +9,11 @@ ci:
   pipeline-gen:
   - build-job:
       script::
-      - - spack compiler find
+      - - export SPACK_USER_CACHE_PATH="${CI_PROJECT_DIR}/_user_cache/"
+        - spack compiler find
         - cd ${SPACK_CONCRETE_ENV_DIR}
         - spack env activate --without-view .
         - if [ -n "$SPACK_BUILD_JOBS" ]; then spack config add "config:build_jobs:$SPACK_BUILD_JOBS"; fi
-        - spack config add "config:install_tree:projections:${SPACK_JOB_SPEC_PKG_NAME}:'morepadding/{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'"
```

Comment:
I don't actually remember why that got removed here, there wasn't anything technical from my point of view... @eugeneswalker was this intentionally removed in the ml-darwin stack or just an oversight?

Reply:
No that is correct. What I mean is I don't know why I did that, I don't remember making that change.

It seems like there was an issue with relocation testing on MacOS (https://github.com/spack/spack/issues/37162)

I re-added it for linux builds here:
https://github.com/spack/spack/pull/38514/files#diff-1c95e7e1793dfb7e3780751007530e7572e944ef124fef1177a9d0e1a4385025R3

Reply:
Thanks.  Just noticing that this was previously done on `cray` as well, so moving into only `linux` changes that.  Any thoughts on how to avoid that without having to duplicate the configuration?  I'm fine just adding into the `cray` configs too though.